src/middleware/packages/ldp/index.js

```diff
@@ -47,74 +62,85 @@ module.exports = {
         accept: 'turtle'
       });
     },
+    /*
+     * Returns a container constructed by the middleware, making a SparQL query on the fly
+     */
     async type(ctx) {
-      ctx.meta.$responseType = 'application/n-triples';
-
-      return await ctx.call('triplestore.query', {
+      let result = await ctx.call('triplestore.query', {
         query: `
-          PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
-          PREFIX as: <https://www.w3.org/ns/activitystreams#>
+          ${this.getPrefixRdf()}
           CONSTRUCT {
           	?suject ?predicate ?object.
           }
           WHERE {
-          	?suject rdf:type ${ctx.params.container} ;
+          	?suject rdf:type ${ctx.params.typeURL} ;
             	?predicate ?object.
           }
               `,
-        accept: 'turtle'
+        accept: 'json'
       });
-    },
-    async subject(ctx) {
-      ctx.meta.$responseType = 'application/n-triples';
+      console.log(result);
+      result = await jsonld.compact(result, this.getPrefixJSON());
+      const { '@graph': graph, '@context': context, ...other } = result;
+
+      const contains = graph == undefined ? (Object.keys(other).length === 0 ? [] : [other]) : graph;
```

Comment:
Cela peut être simplifié comme ça :
```suggestion
      const contains = graph || (Object.keys(other).length === 0 ? [] : [other]);
```